src/Data/ExistsRE.purs

```diff
@@ -0,0 +1,20 @@
+module Data.ExistsRE where
+
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | A variant of `Exists` that works for type constructors that accept a _row_
+-- | of effects.
+data ExistsRE (f :: # ! -> *)
```

Comment:
It's already not possible to pattern match on this - it has no constructor. Essentially this is the same as foreign data import.

Agreed :)